engine/lite/kafka/integration-test/src/it/scala/pl/touk/nussknacker/engine/lite/kafka/NuKafkaRuntimeDockerAvroTest.scala

```diff
@@ -0,0 +1,56 @@
+package pl.touk.nussknacker.engine.lite.kafka
+
+import com.dimafeng.testcontainers._
+import com.typesafe.scalalogging.LazyLogging
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient
+import org.scalatest.{FunSuite, Matchers, TryValues}
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.ConfluentUtils
+import pl.touk.nussknacker.engine.kafka.KafkaZookeeperUtils.richConsumer
+import pl.touk.nussknacker.test.PatientScalaFutures
+
+import java.util.concurrent.TimeoutException
+import scala.util.Try
+
+class NuKafkaRuntimeDockerAvroTest extends FunSuite with BaseNuKafkaRuntimeDockerTest with Matchers with PatientScalaFutures with LazyLogging {
+
+  private val schemaRegistryHostname = "schemaregistry"
+  private val schemaRegistryPort = 8081
+
+  private val schemaRegistryContainer = {
+    val container = GenericContainer(
+      "confluentinc/cp-schema-registry:5.5.0",
+      exposedPorts = Seq(schemaRegistryPort),
+      env = Map(
+        "SCHEMA_REGISTRY_HOST_NAME" -> schemaRegistryHostname,
+        "SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS" -> dockerNetworkKafkaBoostrapServer)
+    )
+    configureNetwork(container, schemaRegistryHostname)
+    container
+  }
+
+  private def mappedSchemaRegistryAddress = s"http://localhost:${schemaRegistryContainer.mappedPort(schemaRegistryPort)}"
+
+  private def dockerNetworkSchemaRegistryAddress = s"http://$schemaRegistryHostname:$schemaRegistryPort"
+
+  override val container: Container = {
+    kafkaContainer.start() // must be started before prepareTestCaseFixture because it creates topic via api
+    schemaRegistryContainer.start() // should be started after kafka
+    fixture = prepareTestCaseFixture("avro-ping-pong", NuKafkaRuntimeTestSamples.avroPingPongScenario)
+    val runtimeContainer = prepareRuntimeContainer(fixture.scenarioFile, Map("SCHEMA_REGISTRY_URL" -> dockerNetworkSchemaRegistryAddress))
+    MultipleContainers(kafkaContainer, schemaRegistryContainer, runtimeContainer)
+  }
+
+  test("avro ping-pong should work") {
+    val schemaRegistryClient = new CachedSchemaRegistryClient(mappedSchemaRegistryAddress, 10)
+    val parsedAvroSchema = ConfluentUtils.convertToAvroSchema(NuKafkaRuntimeTestSamples.avroPingSchema)
+    val inputSchemaId = schemaRegistryClient.register(ConfluentUtils.valueSubject(fixture.inputTopic), parsedAvroSchema)
+    val outputSchemaId = schemaRegistryClient.register(ConfluentUtils.valueSubject(fixture.outputTopic), parsedAvroSchema)
+    val valueBytes = ConfluentUtils.serializeRecordToBytesArray(NuKafkaRuntimeTestSamples.avroPingRecord, inputSchemaId)
+    kafkaClient.sendRawMessage(fixture.inputTopic, "fooKey".getBytes, valueBytes).futureValue
+    consumeFirstError shouldBe empty
```

Comment:
so, it basically waits for 5 secs? I wonder, maybe it would be better to move it after consuming messages and wait shorter (1s?) ? When there is a problem the test will take longer to fail, but it will be shorter in normal circumstances?

Reply:
It won't be so easy - if there will be sth in error, it will be overriden by information that there is timeout waiting on output. I can move it to both `withClue` for checking output and after checking output to won those 5 secs. It won't look too pretty but I've right - time of tests is important

Reply:
ehh `withClue` is not nice unfortunately :-( : 
1. it handles only scaltest exceptions and in case of error we have `TimeoutException` here
2. it evaluates clue eagerly
I've changed 5s -> 1s and added this check in finally block - it is a little bit tricky but don't want to add our own, better `withClue`